src/core/load-dropin.c

```diff
@@ -19,51 +19,245 @@
 
 
 #include "conf-parser.h"
+#include "fs-util.h"
 #include "load-dropin.h"
 #include "load-fragment.h"
 #include "log.h"
+#include "path-util.h"
+#include "stat-util.h"
 #include "strv.h"
 #include "unit-name.h"
 #include "unit.h"
 
+
+typedef struct DependencyConsumerData {
+        Unit *unit;
+        int depth;
+} DependencyConsumerData;
+
+typedef struct DropinSymlink {
+        Unit *unit;
+        UnitDependency dependency;
+        int depth;              /* path depth used when merging 2 dropins */
```

Comment:
"path depth" is basically the directory "level" in which the dropin has been found.

For example, if a dropin is found in :

- /etc/... then the path depth is 1
- /usr/... then the path depth is 4
- etc

Maybe "priority" would be better.

I switched to unsigned.